### ISSUE TITLE
Add mergify bot to the organization

### ISSUE THREAD

**Issue by ccordoba12** [conda-forge/conda-forge-maintenance#49]
https://mergify.io/ is a new web service that automatically merges PRs after CIs are green.

Given the very, very long queues in conda-forge (specially in Appveyor), I think it'd be really handy to add this service to the organization. Else, we have to remember to manually merge PRs for two days or more, and sometimes we forget to do it (it's happened to me a couple of times).

**Comment by CJ-Wright** [comment]
I think it merges things that match a certain criteria (eg passing ci and the PR is from a trusted source). I think this could be an interesting approach, although there are some issues we could run into with this (looking at you ruamel_yaml). Maybe this could be an opt in thing? There may be other ways to help mitigate these issues as well.

**Comment by ccordoba12** [comment]
I haven't used yet but it seems interesting and I just wanted to bring it to your attention.

**Comment by jakirkham** [comment]
Yeah, there's a full config file that one adds per repo with various user specified constraints. We could have `conda-smithy` generate some accepted form of this file.

One can add constraints like [requiring reviews and CIs to pass for example]( https://doc.mergify.io/examples.html#automatic-merge-when-ci-works-and-approving-reviews ). Could also limit these to bot PRs for instance. We'd have to think about what settings we are comfortable with.

**Comment by CJ-Wright** [comment]
It might be good to start out with a small set of packages via an opt in. Not updating the requirements could be an issue with auto-merging (esp if the import tests don't hit everything).

**Comment by isuruf** [comment]
I don't think we should merge automatically, but go for a gitlab style of enabling an automatic merge. For eg: a maintainer can add a `merge` label to the PR after reviewing it and when the tests pass, the bot will merge.

**Comment by ccordoba12** [comment]
> I don't think we should merge automatically

No, not automatically but with a label (as you said) or an approval of at least one of the maintainers.

**Comment by jakirkham** [comment]
I think the point is there are lots of options to affect the kind of constraints we are discussing. So it should be doable. Thanks for raising to our attention @ccordoba12.

On a different note, this repo is at a bare minimum dormant. Should we move this issue to somewhere with higher visibility like the webpage repo? Fortunately GitHub has a beta feature that should let us do this and retain history; so, shouldn't be difficult if it is agreeable.

**Comment by CJ-Wright** [comment]
No objections so I think we can move this.
